docs/style-guide.md

```diff
@@ -33,7 +33,37 @@ Follow the naming advice in [Naming - Rust API Guidelines](https://rust-lang.git
 
 This should result in Crate names always being unambiguously mappable to/from package names (e.g. the Crate "foo_bar" relates to the package "foo-bar").
 
-See also [this discussion on API guidelines](https://github.com/rust-lang/api-guidelines/issues/29#issuecomment-342745898).
+Each component should have its own unique name in form `icu-{package}` which will be used for its releases. When released as part of the meta `icu` package, it will be re-exported as `icu::{package}::{struct}` and for core structures it may also be re-exported as `icu::{struct}}`.
+
+#### Examples
+
+| Package | Crate | Standalone Import | ICU Meta-package | 
+|----|----|----|----|
+| locale | `icu-locale` | `use icu_locale::Locale` | `use icu::Locale` |
+| pluralrules | `icu-pluralrules` | `use icu_pluralrules::PluralRules` | `use icu::PluralRules` |
+| datetime | `icu-datetime` | `use icu_datetime::DateTimeFormat` | `use icu::DateTimeFormat` |
+| datetime | `icu-datetime` | `use icu_datetime::DateTimeStyle` | `use icu::datetime::DateTimeStyle` |
+
+While the scheme may feel repetitive when looking at the import lines, it pays off in being unambigous without aliasing when multiple structs from different components get used together:
+
+```rust
+use icu_locale::Locale;
+use icu_datetime::{DateTimeFormat, options::{DateTimeStyle, DateTimeOptions}};
```

Comment:
What is the `options::` namespace?  You don't mention that above.

Reply:
I was trying to show how specificity may work. `options` are unfortunately not the best example of that, so I switched it to `skeleton`. Hope that works.